pkg/secretsmanager/secretsmanager.go

```diff
@@ -53,6 +65,46 @@ func EnsureSecrets(ctx context.Context, config *types.Configuration, nodes []*ty
 	return nil
 }
 
+var azureVaultURLFmt string = "https://%s.vault.azure.net/"
+
+// newAzureClient create an Azure client with an authorizer from the environment
+func newAzureClient() (*keyvault.BaseClient, error) {
+	authorizer, err := azauth.NewAuthorizerFromEnvironment()
+	if err != nil {
+		return &keyvault.BaseClient{}, err
+	}
+	client := keyvault.New()
+	client.Authorizer = authorizer
+	return &client, nil
+}
+
+// loadAzureSecrets loads any existing secrets in Azure Key Valut into the memory store
+func loadAzureSecrets(ctx context.Context, client *keyvault.BaseClient, vaultName string, nodes []*types.Node) error {
+	for _, node := range nodes {
+		err := loadAzureSecret(ctx, client, vaultName, node)
+		return err
+	}
+	return nil
+}
+
+// loadAzureSecret load a secret from Azure Key Vault
+func loadAzureSecret(ctx context.Context, client *keyvault.BaseClient, vaultName string, node *types.Node) error {
+	secretID := getSecretID(node.SecretConfig.Namespace, node.Path)
+	response, err := client.GetSecret(ctx, fmt.Sprintf(azureVaultURLFmt, vaultName), secretID, "")
+	if err != nil {
+		if e, ok := err.(autorest.DetailedError); ok && e.StatusCode.(int) == 404 {
+			return nil
```

Comment:
@snarlysodboxer I'm a bit new to go but `int(e.StatusCode)`  is an interface, I'm not sure I know another way?
```
pkg/secretsmanager/secretsmanager.go:96:54: cannot convert e.StatusCode (type interface {}) to type int: need type assertion
```

Reply:
Super weird that they make `StatusCode` an interface. Not something I see Google or Kubernetes or other major Go projects doing. Anyway, you're right, you did the right thing with the strange way they wrote the code. It would however panic if for some reason `StatusCode`s value was not an int.

Reply:
yea that was a really fustrating thing to work through, especially with new to go errors.

Reply:
`nil` because the caller of the method is going to make a secret if it's `node.Value == nil`
 the really short is to match the GCPLoadSecret https://github.com/ForgeRock/secret-agent/blob/a8157c6edf1f4e7a1c6061cf03fc81cccc643219/pkg/secretsmanager/secretsmanager.go#L173